experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -0,0 +1,252 @@
+"""
+    get_coupler_config()
+
+Reads in the configuration file and job ID from the command line.
+A dictionary is constructed from the input configuration file, and this
+is returned along with the job ID.
+
+# Returns
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+- `job_id`: A unique identifier for the job
+"""
+function get_coupler_config()
+    ## coupler simulation default configuration
+    parsed_args = parse_commandline(argparse_settings())
+
+    ## modify parsed args for fast testing from REPL #hide
+    if isinteractive()
+        parsed_args["config_file"] =
+            isnothing(parsed_args["config_file"]) ? joinpath(pkg_dir, "config/ci_configs/interactive_debug.yml") :
+            parsed_args["config_file"]
+        parsed_args["job_id"] = "interactive_debug"
+    end
+
+    ## the unique job id should be passed in via the command line
+    job_id = parsed_args["job_id"]
+    @assert !isnothing(job_id) "job_id must be passed in via the command line"
+
+    ## read in config dictionary from file, overriding the coupler defaults in `parsed_args`
+    config_dict = YAML.load_file(parsed_args["config_file"])
+    config_dict = merge(parsed_args, config_dict)
+
+    return config_dict, job_id
+end
+
+"""
+    get_coupler_args!(config_dict)
+
+Extract the necessary arguments from the coupled configuration dictionary.
+This function may modify the input dictionary to remove unnecessary keys.
+
+# Arguments
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+
+# Returns
+- All arguments needed for the coupled simulation
+"""
+function get_coupler_args!(config_dict::Dict)
+    # Simulation-identifying information; Print `config_dict` if requested
+    config_dict["print_config_dict"] && @info(config_dict)
+    mode_name = config_dict["mode_name"]
+
+    # Computational simulation setup information
+    random_seed = config_dict["unique_seed"] ? time_ns() : 1234
+    FT = config_dict["FLOAT_TYPE"] == "Float64" ? Float64 : Float32
+    comms_ctx = Utilities.get_comms_context(config_dict)
+
+    # Time information
+    t_end = Float64(Utilities.time_to_seconds(config_dict["t_end"]))
+    t_start = Float64(Utilities.time_to_seconds(config_dict["t_start"]))
+    date0 = date = Dates.DateTime(config_dict["start_date"], Dates.dateformat"yyyymmdd")
+    Δt_cpl = Float64(Utilities.time_to_seconds(config_dict["dt_cpl"]))
+    component_dt_dict = parse_component_dts!(config_dict)
```

Comment:
I'd advise against having mutating functions that return something. It's harder to know what we should expect from them. Either they return something, or they mutate the argument. If you want the return value, you can grab it in a second function that only grabs the return value